src/App.tsx

```diff
@@ -1,156 +1,115 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
+import { getTodos } from './api/todos';
+import { Todo } from './types/Todo';
+import { Selected } from './types/Selected';
 import { UserWarning } from './UserWarning';
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList';
+import { FilterTodos } from './components/FilterTodos';
+import { ErrorMessage } from './components/ErrorMessage';
+import { ErrorMessages } from './types/ErrorMessages';
+
+const USER_ID = 11281;
+
+function getPreparedTodos(
+  todos: Todo[], selected: Selected,
+) {
+  let preparedTodos = [...todos];
+
+  if (selected) {
+    preparedTodos = preparedTodos.filter(todo => {
+      switch (selected) {
+        case Selected.ACTIVE: {
+          return !todo.completed;
+        }
+
+        case Selected.COMPLETED: {
+          return todo.completed;
+        }
+
+        default:
+          return true;
+      }
+    });
+  }
```

Comment:
No need to create copy - filter method do not mutate original array. No need to filter in case Selected is all - move filter call inside case. In default case return todos